code/worths_test/~worths_test_subsystems.dm

```diff
@@ -0,0 +1,282 @@
+		/obj/item/gun/energy/gun/secure/mounted,
+		/obj/item/crafting_holder,
+		/obj/item/pressure,
+		/obj/machinery/portable_atmospherics/hydroponics/soil/invisible,
+		/obj/item/reagent_containers/pill/pouch_pill,
+		/obj/machinery/power/smes,
+		/obj/turbolift_map_holder,
+		/obj/item/material,
+		/obj/item/fuel_assembly,
+		/obj/item/reagent_containers/glass/mister,
+		/obj/item/clothing/suit/storage/hooded,
+		/obj/hotspot
+	)
+
+/datum/controller/subsystem/worths_test/Initialize(start_uptime)
+	#ifndef UNIT_TEST_COLOURED
+	if(world.system_type != UNIX) // Not a Unix/Linux/etc system, we probably don't want to print color escapes (unless UNIT_TEST_COLOURED was defined to force escapes)
+		ascii_esc = ""
+		ascii_red = ""
+		ascii_green = ""
+		ascii_yellow = ""
+		ascii_reset = ""
+	#endif
+
+	log_worths_test("Initializing Worths Testing")
+
+	src.processed_materials = list()
+	src.processed_objects = list()
+	src.traders = list()
+
+	for(var/path in src.root_exceptions)
+		src.exceptions |= typesof(path)
+
+	SSticker.master_mode = "extended"
+
+/datum/controller/subsystem/worths_test/proc/start_game()
+	if (GAME_STATE >= RUNLEVEL_POSTGAME)
+		log_worths_test("Unable to start testing - game is finished!")
+		del world
+		return
+
+	if ((GAME_STATE == RUNLEVEL_LOBBY) && !SSticker.start_now())
+		log_worths_test("Unable to start testing - SSticker failed to start the game!")
+		del world
+		return
+
+	stage++
+	log_worths_test("Game start has been requested.")
+
+/datum/controller/subsystem/worths_test/proc/await_game_running()
+	if(GAME_STATE == RUNLEVEL_GAME)
+		log_worths_test("The game is now in progress.")
+		stage++
+
+/datum/controller/subsystem/worths_test/proc/init_material_queue()
+	src.material_queue = list()
+
+	for(var/material/mat in SSmaterials.materials)
+		if((mat.type in src.processed_materials) || istype(mat, /material/placeholder))
+			continue
+		src.material_queue.Add(mat)
+
+	if(length(src.material_queue))
+		src.new_materials = TRUE
+
+	stage++
+
+/datum/controller/subsystem/worths_test/proc/process_materials()
+	while(length(src.material_queue))
+		var/material/mat = src.material_queue[length(src.material_queue)]
+		LIST_DEC(src.material_queue)
+
+		log_worths_test("**** [uppertext(mat.name)] ****")
+
+		var/list/base_recipes = mat.dump_recipes()
+		var/list/reinforced_recipes = list()
+
+		var/mat_name = lowertext(mat.name)
+		log_worths_test("-- [mat_name]: [length(base_recipes)] recipie\s")
+
+		for(var/material/reinf_mat in SSmaterials.materials)
+			if(istype(reinf_mat, /material/placeholder))
+				continue
+
+			var/list/recipes = mat.dump_recipes(reinf_mat)
+
+			if(!length(recipes))
+				continue
+
+			log_worths_test("-- [mat_name] (Reinforced - [lowertext(reinf_mat.name)]): [length(recipes)] recipie\s")
+
+			reinforced_recipes[lowertext(reinf_mat.name)] = recipes
+
+		src.processed_materials[mat.type] = list(
+			"id" = mat_name,
+			"base recipes" = base_recipes,
+			"reinforced recipes" = reinforced_recipes
```

Comment:
```suggestion
			"base_recipes" = base_recipes,
			"reinforced_recipes" = reinforced_recipes
```
Easier to parse this way

Reply:
No idea why I didn't do that here, I did for the materials.  I blame coding at 3am